tests/gtfs/test_validation.py

```diff
@@ -580,3 +602,233 @@ def test_summarise_routes_on_pass(self, gtfs_fixture):
             "Size of date_route_counts not as expected. "
             "Expected {expected_size}"
         )
+
+    def test__plot_summary_defences(self, gtfs_fixture):
+        """Test defences for _plot_summary()."""
+        current_fixture = gtfs_fixture
+        current_fixture.summarise_routes()
+
+        # test parameters that are yet to be tested
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "'orientation expected 'v' or 'h'"
+                " (non case sensitive)."
+                " Found l of type <class 'str'>."
+            ),
+        ):
+            current_fixture._plot_summary(
+                current_fixture.daily_route_summary,
+                "route_count_mean",
+                orientation="l",
+            )
+
+    @pytest.mark.filterwarnings("ignore::UserWarning")
```

Comment:
@CBROWN-ONS this one needs some thought. Ideally, we want to test the behaviour of these warnings rather than filter them. Did you try it? I have found this to be unexpectedly hacky in Python and if it's way too much work to implement, will potentially spin a separate issue up.

Reply:
The main rationale for this was that the functionality of saving both an image and a html file at one time would always raise a warning as you are guaranteed not to have specified a correct file type (e.g., .PNG) for both html and image saves. This only appears as a UserWarning to inform the user, however the overall functionality remains consistent and file extensions will always be adjusted to a valid type.  
Without filtering these warnings, they would show up in the test report, indicating that there was a warning  (or something wrong with tests) when in reality there wasn't any major issue.  
I'm not sure how to approach this but it's possible that the solution I mentioned in the above comment may help to resolve this.